homeassistant/components/sensor/launch.py

```diff
@@ -0,0 +1,92 @@
+"""
+A sensor platform that give you information about the next space launch.
+
+For more details about this platform, please refer to the documentation at
+https://www.home-assistant.io/components/sensor.launch/
+"""
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import ATTR_ATTRIBUTION, CONF_NAME
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+REQUIREMENTS = ['pylaunches==0.1.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_ATTRIBUTION = "Data provided by Launch Library."
+
+DEFAULT_NAME = 'Launch'
+
+SCAN_INTERVAL = timedelta(hours=1)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    })
+
+
+async def async_setup_platform(
+        hass, config, async_add_entities, discovery_info=None):
+    """Create the launch sensor."""
+    from pylaunches.api import Launches
+
+    name = config[CONF_NAME]
+
+    session = async_get_clientsession(hass)
+    launches = Launches(hass.loop, session)
+    sensor = [LaunchSensor(launches, name)]
+    async_add_entities(sensor, True)
+
+
+class LaunchSensor(Entity):
+    """Representation of a launch Sensor."""
+
+    def __init__(self, launches, name):
+        """Initialize the sensor."""
+        self.launches = launches
+        self._attributes = {}
+        self._name = name
+        self._state = None
+
+    async def async_update(self):
+        """Get the latest data."""
+        await self.launches.get_launches()
+        if self.launches.launches is None:
+            _LOGGER.error("No data recieved.")
+            return
+        try:
+            data = self.launches.launches[0]
+            self._state = data['name']
+            self._attributes['launch_time'] = data['start']
+            self._attributes['agency'] = data['agency']
+            self._attributes['agency_country_code'] = (data
+                                                       ['agency_country_code'])
```

Comment:
Please don't break the line before the bracket. Directly after the opening bracket is ok, or directly after the first parenthesis.

The preferred way of breaking lines according to PEP8 is within parenthesis, brackets and braces:
https://www.python.org/dev/peps/pep-0008/#maximum-line-length